yb-voyager/cmd/exportData.go

```diff
@@ -735,50 +747,105 @@ func reportUnsupportedTables(finalTableList []sqlname.NameTuple) {
+	sourceTypeHandlesPartitionAsSeparateTable := func(dbType string) bool {
+		return dbType == POSTGRESQL || dbType == YUGABYTEDB
+	}
+	if !sourceTypeHandlesPartitionAsSeparateTable(source.DBType) || goToNameRegDirectly {
+		//ORACLE and MySQL no need to care about leaf partitions
+		tuple, err := namereg.NameReg.LookupTableName(qualifiedObjectStr)
+		if err != nil {
+			return sqlname.NameTuple{}, fmt.Errorf("lookup for table name failed err: %s: %v", qualifiedObjectStr, err)
+		}
+		return tuple, nil
+	}
+	//Now for PG/YB create a ObjectName and a NameTuple by hand and then check if that is a partition table or not
+	var obj *sqlname.ObjectName
+	if qualifiedObjectName != nil {
+		//if passed in parameter then take it else create one
+		obj = qualifiedObjectName
+	} else {
+		//create the ObjectName by hand
 		defaultSchemaName, _ := getDefaultSourceSchemaName()
-		//For partitions case there is no defined mapping and
-		//hence lookup will fail, need to create nametuple for non-root table by hand
-		obj := sqlname.NewObjectName(source.DBType, defaultSchemaName, schema, table)
-		tuple := sqlname.NameTuple{
-			SourceName:  obj,
-			CurrentName: obj,
-		}
-		parent := ""
-		if source.DBType == POSTGRESQL || source.DBType == YUGABYTEDB {
-			parent = source.DB().ParentTableOfPartition(tuple)
-		}
-		if parent == "" {
-			tuple, err = namereg.NameReg.LookupTableName(fmt.Sprintf("%s.%s", schema, table))
-			if err != nil {
-				utils.ErrExit("lookup for table name failed err: %s: %v", table, err)
-			}
+		obj = sqlname.NewObjectNameWithQualifiedName(source.DBType, defaultSchemaName, qualifiedObjectStr)
+	}
+	//get the name tupe case its leaf partition return a handcrafted NameTuple else return  from nameReg lookup
+	var err error
+	tuple := sqlname.NameTuple{
+		SourceName:  obj,
+		CurrentName: obj,
+	}
+	parent := source.DB().ParentTableOfPartition(tuple)
+
+	if parent == "" {
+		tuple, err = namereg.NameReg.LookupTableName(fmt.Sprintf("%s.%s", obj.SchemaName, obj.Unqualified.Unquoted))
+		if err != nil {
+			return sqlname.NameTuple{}, fmt.Errorf("lookup for table name failed err: %s: %v", obj.Unqualified, err)
 		}
-		fullTableList = append(fullTableList, tuple)
 	}
-	excludeTableList := extractTableListFromString(fullTableList, source.ExcludeTableList, "exclude")
-	if len(excludeTableList) > 0 {
-		//TODO: avoid duplicate call to this function and optimize this function later
-		_, excludeTableList, err = addLeafPartitionsInTableList(excludeTableList, true)
+	return tuple, nil
+}
+
+func applyTableListFlagsOnFullListAndAddLeafPartitions(fullTableList []sqlname.NameTuple, tableListViaFlag string, excludeTableListViaFlag string) ([]sqlname.NameTuple, error) {
+	var err error
+	var includeTableList, excludeTableList []sqlname.NameTuple
+
+	applyFilterAndAddLeafTable := func(flagList string, flagName string) ([]sqlname.NameTuple, error) {
```

Comment:
This structure of applying include/exclude is relatively easy to understand, thanks 👍